src/main/java/fi/aalto/cs/intellij/actions/InstallPluginsNotificationAction.java

```diff
@@ -0,0 +1,128 @@
+  }
+
+  /**
+   * Builds the action.
+   *
+   * @param missingIdeaPluginDescriptors is a {@link List} of {@link IdeaPluginDescriptor} that can
+   *                                     be installed.
+   */
+  public InstallPluginsNotificationAction(
+      List<IdeaPluginDescriptor> missingIdeaPluginDescriptors,
+      PluginInstaller pluginInstaller,
+      RestartProposer restartProposer) {
+    super("Install missing ("
+        + getPluginsNamesString(missingIdeaPluginDescriptors) + ") plugin(s).");
+    this.missingIdeaPluginDescriptors = missingIdeaPluginDescriptors;
+    this.pluginInstaller = pluginInstaller;
+    this.restartProposer = restartProposer;
+  }
+
+  /**
+   * Install missing plugins and propose a restart.
+   *
+   * @param e            an {@link AnActionEvent}.
+   * @param notification a {@link Notification} to handle.
+   */
+  @Override
+  public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
+    //todo what if all the plugins are dynamic? (no way, I know)
+    // what if the plugins are not installed at all?
+    missingIdeaPluginDescriptors.forEach(descriptor -> {
+      try {
+        pluginInstaller.install(descriptor);
+      } catch (IOException ex) {
+        logger.error("Could not install plugin" + descriptor.getName() + ".", ex);
+      }
+    });
+    notification.expire();
+    restartProposer.proposeRestart();
+  }
+
+  /**
+   * A method responsible for plugin installation.
+   */
+  public static void install(IdeaPluginDescriptor descriptor) throws IOException {
+    PluginDownloader pluginDownloader = PluginDownloader.createDownloader(descriptor);
+    pluginDownloader.prepareToInstall(new BgProgressIndicator());
+    pluginDownloader.install();
+  }
+
+  /**
+   * A method to propose a restart when the plugins have been installed.
+   */
+  public static void proposeRestart() {
+    PluginManagerConfigurable
+        .shutdownOrRestartApp("Plugins required for A+ course are now installed.");
+  }
+
+  /**
+   * An abstract interface for an object that installs plugins based on the {@link
+   * IdeaPluginDescriptor} provided. The most useful realization of this interface is {@code
+   * InstallPluginsNotificationAction::install}.
+   */
+  @FunctionalInterface
+  public interface PluginInstaller {
+
+    /**
+     * Installs a given {@link IdeaPluginDescriptor}.
+     *
+     * @param descriptor is the {@link IdeaPluginDescriptor} of the desired plugin.
+     */
+    void install(IdeaPluginDescriptor descriptor) throws IOException;
+  }
+
+  /**
+   * An abstract interface for an object that proposes the IDE restart. The most useful realization
+   * of this interface is {@code InstallPluginsNotificationAction::proposeRestart}.
+   */
+  @FunctionalInterface
+  public interface RestartProposer {
+
+    /**
+     * Calls for an IJ framework to trigger a restart of the IDE.
+     */
+    void proposeRestart();
+  }
```

Comment:
I find this a nice enough solution at least for now. If the framework interface of this class gets wider from this in the future, we can consider constructing a single facade of these.